parlai/crowdsourcing/tasks/fast_acute/analysis.py

```diff
@@ -0,0 +1,743 @@
+from mephisto.data_model.unit import Unit as MephistoUnit
+from scipy.stats import binom_test
+
+from parlai.core.params import ParlaiParser
+from parlai.crowdsourcing.tasks.acute_eval.acute_eval_blueprint import BLUEPRINT_TYPE
+
+# To register the ACUTE-Eval blueprint
+from parlai.crowdsourcing.tasks.fast_acute.util import get_hashed_combo_path
+
+_ = BLUEPRINT_TYPE
+# NOTE: BLUEPRINT_TYPE needs to be imported here to register the blueprint
+
+# throw away turkers below this threshold
+AGREEMENT_THRESHOLD = 0.8
+# do we count ties as agreements?
+AGREEMENT_TIES_OKAY = False
+# NOTE: these could be added as flags if desired
+
+# Prepended to checkbox columns in self.dataframe
+CHECKBOX_PREFIX = 'checkbox: '
+
+
+def setup_args():
+    """
+    Setup appropriate args.
+    """
+    parser = ParlaiParser(False, False)
+    parser.add_argument(
+        '-id', '--run-id', type=str, default=None, help='run id to analyze'
+    )
+    parser.add_argument(
+        '--root-dir', type=str, default=None, help='root ACUTE-Eval save directory'
+    )
+    parser.add_argument(
+        '--outdir', type=str, default=None, help='where to save the results'
+    )
+    parser.add_argument(
+        '--mephisto-root',
+        type=str,
+        default=None,
+        help='Where to check for mephisto data (default own dir)',
+    )
+    parser.add_argument(
+        '--model-ordering',
+        type=str,
+        default=None,
+        help='Comma-separated list of models, in the order in which to display them',
+    )
+    return parser
+
+
+class AcuteAnalyzer(object):
+    """
+    Analyzer.
+
+    Given a run_id, we can do lots of fun things!
+    """
+
+    def __init__(self, opt: Dict, remove_failed: bool = True):
+        """
+        Initialize the analyzer.
+
+        Builds up the dataframe
+
+        :param opt:
+            opt dict
+
+        :param remove_failed:
+            Whether to remove ratings from turkers who failed onboarding
+        """
+        self.root_dir = opt['root_dir']
+        assert os.path.isdir(self.root_dir), '--root-dir must be a real directory!'
+        self.run_id = opt['run_id']
+        self.outdir = opt['outdir']
+        # Get task for loading pairing files
+        self.task = opt.get('task', 'q')
+        if opt.get('model_ordering') is not None:
+            self.custom_model_ordering = opt['model_ordering'].split(',')
+        else:
+            self.custom_model_ordering = None
+        if not self.outdir:
+            self.outdir = os.path.join(self.root_dir, f'{self.run_id}-results')
+        if not os.path.exists(self.root_dir):
+            os.mkdir(self.root_dir)
+        if not os.path.exists(self.outdir):
+            os.mkdir(self.outdir)
+        mephisto_root_path = opt['mephisto_root']
+        if not mephisto_root_path:
+            mephisto_root_path = None
+        mephisto_db = LocalMephistoDB(database_path=mephisto_root_path)
```

Comment:
Heads up that this path doesn't match where the Mephisto DB lives. 

(Was using this file as a base for some analysis and had to change this line to `mephisto_db = LocalMephistoDB(database_path=mephisto_root_path + "/data/database.db")`)

Reply:
Hmm - instead of appending `"/data/database.db"` to `mephisto_root_path`, you should be able to just append it to your `--mephisto-root` flag instead